src/pru-motion-queue.cc

```diff
@@ -79,6 +81,53 @@ static void DumpMotionSegment(volatile const struct MotionSegment *e,
 }
 #endif
 
+// Store the required informations needed to backtrack the absolute position.
+struct HistorySegment {
+  uint32_t fractions[MOTION_MOTOR_COUNT];
+  uint32_t cumulative_loops[MOTION_MOTOR_COUNT];
+  uint8_t direction_bits;
+};
+
+void PRUMotionQueue::RegisterHistorySegment(const MotionSegment *element) {
```

Comment:
If we're not modifying the MotionSegment, maybe we should pass it via const reference.

My rules I use for pointer vs. reference:
  Pointer: when we want to modify something.
  const Pointer: when we don't want to modify things, but if the parameter can be nullptr; so in that case we need to distinguish if if it is there or not.
  const Reference: when we don't want to modify things and if the thing is always there (so cannot be null).

(regular reference as a parameter without const: almost never: it has too surprising effects.)